From 4a2a0cbe058d10ae23ce03f387e638fb607cbf56 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Tue, 9 Apr 2013 14:16:10 +0800
Subject: [PATCH 12/37] ENGR00256175 Android reboot if setup multi-user with
 space running out

Convert the BlockSize and AvailableBlocks to be long(64bit) for those case total
available space is more than 2G.

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 services/java/com/android/server/pm/UserManagerService.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/java/com/android/server/pm/UserManagerService.java b/services/java/com/android/server/pm/UserManagerService.java
index 24ca318..c13d5b1 100644
--- a/services/java/com/android/server/pm/UserManagerService.java
+++ b/services/java/com/android/server/pm/UserManagerService.java
@@ -439,7 +439,7 @@ public class UserManagerService extends IUserManager.Stub {
     private static final boolean isNoFreeSpace() {
         final String path = Environment.getDataDirectory().getPath();
         StatFs fileStats = new StatFs(path);
-        long restM = fileStats.getAvailableBlocks() * fileStats.getBlockSize();
+        long restM = (long) fileStats.getAvailableBlocks() * (long) fileStats.getBlockSize();
         Slog.w(LOG_TAG, "remain size for data is " + restM);
         if ( restM < MIN_AVAILABLE_SPACE)
             return true;
-- 
1.8.0

